lib/Interpreter/DeclUnloader.cpp

```diff
@@ -711,10 +736,49 @@ bool DeclUnloader::VisitRedeclarable(clang::Redeclarable<T>* R, DeclContext* DC)
 
   bool DeclUnloader::VisitNamespaceDecl(NamespaceDecl* NSD) {
     // NamespaceDecl: NamedDecl, DeclContext, Redeclarable
-    bool Successful = VisitDeclContext(NSD);
-    Successful &= VisitRedeclarable(NSD, NSD->getDeclContext());
+
+    // When unloading a NamedDecl within an inline namespace within std the
+    // NamedDecl needs to be removed from the parent std as well.
+    // Not sure why, but test/CodeUnloading/NameSpaces.C demonstrates
```

Comment:
The tests don't rely on any std headers, just use lines 12-24 that tests the issue.
Later in that file there is an unloading of an included `std except` but that just tests the larger issue that exposed the problem.

My point is that it seems you are speculating what I've said is not right. Its up to you to proove that this is in fact more general of an issue as I've done everything I could and came to the conclusion **IT ONLY AFFECTS STD!**